randovania/exporter/game_exporter.py

```diff
@@ -58,18 +67,67 @@ def export_game(
         progress_update: status_update_lib.ProgressUpdateCallable,
     ):
         meta_data: PatcherDataMeta = patch_data.pop("_randovania_meta")
-        self._before_export()
-
-        try:
-            with sentry_sdk.configure_scope() as scope:
-                scope.add_attachment(
-                    json.dumps(patch_data).encode("utf-8"),
-                    filename="patcher.json",
-                    content_type="application/json",
-                )
-                with scope.start_transaction(op="task", name="export_game") as span:
-                    span.set_tag("exporter", type(self).__name__)
-                    span.set_tag("layout_was_user_modified", meta_data["layout_was_user_modified"])
+        with sentry_sdk.isolation_scope() as scope:
+            scope.add_attachment(
+                json.dumps(patch_data).encode("utf-8"),
+                filename="patcher.json",
+                content_type="application/json",
+                add_to_transactions=True,
+            )
+            with scope.start_transaction(op="task", name="export_game") as span:
+                span.set_tag("exporter", type(self).__name__)
+                span.set_tag("layout_was_user_modified", meta_data["layout_was_user_modified"])
+
+                self._before_export()
+                try:
                     self._do_export_game(patch_data, export_params, progress_update)
-        finally:
-            self._after_export()
+
+                except AbortBackgroundTask:
+                    raise
+
+                except Exception as e:
+                    input_hash = export_params.calculate_input_hash()
+                    scope.capture_exception(e, contexts=input_hash)
+                    # TODO: compare `input_hash` with known good values and if different,
+                    # wrap into a "User Error" exception
+                    raise
+
+                finally:
+                    self._after_export()
+
+
+def _hash_file_internal(path: Path) -> typing.Any:
+    sha = hashlib.sha1()
+
+    with path.open("rb") as f:
+        while True:
+            data = f.read(65536)
```

Comment:
why this number specifically?

Reply:
random example I found online. Seemed like a decent chunk to read, being 64kb

Reply:
If we only read 64kb, isnt the system then useless for e.g the gamecube games where there were dumping errors happening after 64kb?

Reply:
There's a `while True`?

Reply:
nvm whoops.